judo-requirement-report-processor/src/main/java/hu/blackbelt/judo/requirement/report/processor/RequirementProcessor.java

```diff
@@ -0,0 +1,151 @@
+package hu.blackbelt.judo.requirement.report.processor;
+
+/*-
+ * #%L
+ * JUDO Requirement :: Report :: Processor
+ * %%
+ * Copyright (C) 2018 - 2023 BlackBelt Technology
+ * %%
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License, v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is
+ * available at https://www.gnu.org/software/classpath/license.html.
+ * 
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ * #L%
+ */
+
+import com.google.auto.service.AutoService;
+import com.opencsv.*;
+import org.junit.jupiter.api.Test;
+import javax.annotation.processing.*;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.TypeElement;
+import javax.tools.Diagnostic;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.*;
+import java.util.stream.Collectors;
+import hu.blackbelt.judo.requirement.report.annotation.Requirement;
+
+@SupportedAnnotationTypes(
+        "hu.blackbelt.judo.requirement.report.annotation.Requirement"
+)
+@SupportedSourceVersion(SourceVersion.RELEASE_11)
+@AutoService(Processor.class)
+public class RequirementProcessor extends AbstractProcessor {
+
+    public RequirementProcessor(){}
+
+    @Override
+    public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
+        processingEnv.getMessager().printMessage(
+                Diagnostic.Kind.NOTE,
+                "RequirementProcessor start."
+        );
+
+        if (annotations.isEmpty()) {
+            return false;
+        }
+
+        // Annotation[] -> Element[] -> Info[]
+        Collection<Info> infos = annotations.stream()
+                .flatMap(a -> roundEnv.getElementsAnnotatedWith(a).stream())
+                .collect(Collectors.toSet()).stream()
+                .flatMap(e -> collectReqForElement(e).stream())
+                .collect(Collectors.toSet());
+
+        writeCsv(new File(processingEnv.getOptions().get("reportPath")), infos);
+
+        processingEnv.getMessager().printMessage(
+                Diagnostic.Kind.NOTE,
+                "RequirementProcessor end."
+        );
+        return false;
```

Comment:
Does this function always return false?

Reply:
Yes. Because this processor doesn't want to stop other annotation processors.
"if false is returned, the annotation types are unclaimed and subsequent processors may be asked to process them."